src/radical/pilot/utils/session.py

```diff
@@ -111,35 +110,47 @@ def fetch_profiles (sid, dburl=None, src=None, tgt=None, access=None,
 
             # Try to fetch a tarball of profiles, so that we can get them
             # all in one (SAGA) go!
-            PROFILES_TARBALL = '%s.prof.tgz' % pilot['uid']
+            PROFILES_TARBALL = '%s.prof.tbz' % pilot['uid']
             tarball_available = False
             try:
                 if  sandbox.is_file(PROFILES_TARBALL) and \
-                    sandbox.get_size(PROFILES_TARBALL):
-
+                        sandbox.get_size(PROFILES_TARBALL):
                     log.info("profiles tarball exists")
-                    ftgt = rs.Url('%s/%s' % (tgt_url, PROFILES_TARBALL))
-
-                    if skip_existing and os.path.isfile(ftgt.path) \
-                            and os.stat(ftgt.path).st_size > 0:
-
-                        log.info("skip fetching of '%s/%s' to '%s'.",
-                                 sandbox_url, PROFILES_TARBALL, tgt_url)
-                        tarball_available = True
-                    else:
-
-                        log.info("fetch '%s%s' to '%s'.", sandbox_url,
-                                 PROFILES_TARBALL, tgt_url)
-
-                        prof_file = rs.fs.File("%s%s" % (sandbox_url,
-                                            PROFILES_TARBALL), session=session)
-                        prof_file.copy(ftgt, flags=rs.fs.CREATE_PARENTS)
-                        prof_file.close()
+                else:
+                    log.warn("profiles tarball doesnt exists! creating now.")
+                    # so lets create a tarball right now with SAGA JobService
+                    js_url = pilot['js_hop']
+                    log.debug('js  : %s', js_url)
+                    js = rs.job.Service(js_url, session=session)
+
+                    cmd = "cd %s; find . -name \\*.prof > profiles.lst; " \
+                          "tar cjf %s -T profiles.lst" % (sandbox.url.path, PROFILES_TARBALL)
+                    j = js.run_job(cmd)
+                    j.wait()
+                    log.debug('tarball cmd for profiles : %s', cmd)
+                    log.debug('tar result : %s\n---\n%s\n---\n%s',
+                              j.get_stdout_string(), j.get_stderr_string(),
+                              j.exit_code)
+                    if not j.exit_code:
+                        log.info("profiles tarball created successfully")
+
+                ftgt = rs.Url('%s/%s' % (tgt_url, PROFILES_TARBALL))
```

Comment:
This is probably inconsistent throughout the code, but please switch from `rs.Url` to `ru.Url`.  The `Url` class was moved to RU a while back.